core/validator.go

```diff
@@ -43,74 +106,108 @@ func (v *chunkValidator) ValidateBlob(blob *BlobMessage, operatorState *Operator
-		if err != nil {
-			return err
-		}
+func (v *chunkValidator) UpdateOperatorID(operatorID OperatorID) {
+	v.operatorID = operatorID
+}
 
-		// Validate the chunkLength against the quorum and adversary threshold parameters
-		numOperators := uint(len(operatorState.Operators[quorumHeader.QuorumID]))
-		minChunkLength, err := v.assignment.GetMinimumChunkLength(numOperators, blob.BlobHeader.BlobCommitments.Length, quorumHeader.QuantizationFactor, quorumHeader.QuorumThreshold, quorumHeader.AdversaryThreshold)
-		if err != nil {
-			return err
+func (v *chunkValidator) ValidateBatch(blobs []*BlobMessage, operatorState *OperatorState) error {
+	subBatchMap := make(map[EncodingParams]*SubBatch)
+
+	for _, blob := range blobs {
+		if len(blob.Bundles) != len(blob.BlobHeader.QuorumInfos) {
+			return errors.New("number of bundles does not match number of quorums")
 		}
-		params, err := GetEncodingParams(minChunkLength, info.TotalChunks)
+
+		// Validate the blob length
+		err := v.encoder.VerifyBlobLength(blob.BlobHeader.BlobCommitments)
 		if err != nil {
 			return err
 		}
-
-		if params.ChunkLength != chunkLength {
-			return errors.New("number of chunks does not match assignment")
-		}
-
-		// Get the chunk length
-		chunks := blob.Bundles[quorumHeader.QuorumID]
-		for _, chunk := range chunks {
-			if uint(chunk.Length()) != chunkLength {
-				return ErrChunkLengthMismatch
+		// for each quorum
+		for _, quorumHeader := range blob.BlobHeader.QuorumInfos {
+			// Check if the operator is a member of the quorum
+			chunks, assignment, params, err := v.preprocessBlob(quorumHeader, blob, operatorState)
+			if err == ErrBlobQuorumSkip {
+				continue
+			} else if err != nil {
+				return err
+			} else {
+				// Check the received chunks against the commitment
+				blobIndex := 0
+				subBatch, ok := subBatchMap[*params]
+				if ok {
+					blobIndex = subBatch.NumBlobs
+				}
+
+				indices := assignment.GetIndices()
+				samples := make([]Sample, len(chunks))
+				for ind := range chunks {
+					samples[ind] = Sample{
+						Commitment:      blob.BlobHeader.BlobCommitments.Commitment,
+						Chunk:           chunks[ind],
+						AssignmentIndex: uint(indices[ind]),
+						BlobIndex:       blobIndex,
+					}
+				}
+
+				// update subBatch
+				if !ok {
+					subBatchMap[*params] = &SubBatch{
+						Samples:  samples,
+						NumBlobs: 1,
+					}
+				} else {
+					subBatch.Samples = append(subBatch.Samples, samples...)
+					subBatch.NumBlobs += 1
+				}
 			}
 		}
+	}
 
-		// Validate the chunk length
-		if chunkLength*quorumHeader.QuantizationFactor*numOperators != quorumHeader.EncodedBlobLength {
-			return ErrInvalidHeader
-		}
+	// Parallelize the universal verification for each subBatch
+	numSubBatch := len(subBatchMap)
+	out := make(chan error, numSubBatch)
+	for params, subBatch := range subBatchMap {
+		params := params
+		subBatch := subBatch
+		go v.universalVerifyWorker(params, subBatch, out)
```

Comment:
I agree that it's better to cap the threading with a pool.
I don't have strong objection to use a threadpool here. But if that's a concern, an alternative is to make the core lib to create the SubBatch map, and the Node to run a threadpool to invoke the execution for each SubBatch.

Reply:
for the second approach, we need to export the encoder interface from the validator. That will need discussion and a proper code restructure.  I will choose option 1. @mooselumph what do you think

Reply:
I don't think the refactoring would be needed... just give the validator a public method to be called by the node worker. But I don't really have a strong opinion about whether the threadpool should go here or in the node.